src/lightning_app/components/database/server.py

```diff
@@ -0,0 +1,138 @@
+        db_filename: str = "database.db",
+        debug: bool = False,
+    ) -> None:
+        """The Database Component enables to interact with an SQLite database to store some structured information
+        about your application.
+
+        The provided models are SQLModel tables
+
+        Arguments:
+            models: A SQLModel or a list of SQLModels table to be added to the database.
+            db_filename: The name of the SQLite database.
+            debug: Whether to run the database in debug mode.
+            mode: Whether the database should be running within the flow or dedicated work.
+
+        Example:
+
+            from sqlmodel import SQLModel, Field
+            from lightning import LightningFlow, LightningApp
+            from lightning_app.components.database import Database, DatabaseClient
+
+            class CounterModel(SQLModel, table=True):
+                __table_args__ = {"extend_existing": True}
+
+                id: int = Field(default=None, primary_key=True)
+                count: int
+
+
+            class Flow(LightningFlow):
+
+                def __init__(self):
+                    super().__init__()
+                    self.db = Database(models=[CounterModel])
+                    self._client = None
+                    self.counter = 0
+
+                def run(self):
+                    self.db.run()
+
+                    if not self.db.alive():
+                        return
+
+                    if self.counter == 0:
+                        self._client = DatabaseClient(model=CounterModel, db_url=self.db.url)
+                        self._client.reset_database()
+
+                    assert self._client
+
+                    rows = self._client.select_all()
+
+                    print(f"{self.counter}: {rows}")
+
+                    if not rows:
+                        self._client.insert(CounterModel(count=0))
+                    else:
+                        row: CounterModel = rows[0]
+                        row.count += 1
+                        self._client.update(row)
+
+                    if self.counter >= 100:
+                        row: CounterModel = rows[0]
+                        self._client.delete(row)
+                        self._client.delete_database()
+                        self._exit()
+
+                    self.counter += 1
+
+            app = LightningApp(Flow())
+        """
+        super().__init__(parallel=True, cloud_build_config=BuildConfig(["sqlmodel"]))
+        self.db_filename = db_filename
+        self.debug = debug
+        self._models = models if isinstance(models, list) else [models]
+        self._client = None
+
+    def run(self) -> None:
+        app = FastAPI()
+
+        create_database(self.db_filename, self._models, self.debug)
+        models = {m.__name__: m for m in self._models}
+        app.post("/select_all/")(partial(general_select_all, models=models))
```

Comment:
This looks great, the fundamental hold-up I have is that we are exposing raw database methods to the outside world. This may be ok for a trusted environment, but it's not ok for an untrusted one.
We should restrict the origin of the requests to allow requests from the app itself, it should be easily doable.

Reply:
Add a security mechanism with a token. There is a period of time where the token can be accessed from the AppState until the work remove it from its state.